pkg/reconciler/metrics/metrics.go

```diff
@@ -0,0 +1,240 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	triggersclient "github.com/tektoncd/triggers/pkg/client/injection/client"
+
+	triggersclientset "github.com/tektoncd/triggers/pkg/client/clientset/versioned"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/metrics"
+)
+
+var (
+	elMetricsName = "eventlistener_count"
+	elCount       = stats.Float64(elMetricsName,
+		"number of eventlistener",
+		stats.UnitDimensionless)
+	elCountView *view.View
+
+	tbMetricsName = "triggerbinding_count"
+	tbCount       = stats.Float64(tbMetricsName,
+		"number of triggerbinding",
+		stats.UnitDimensionless)
+	tbCountView *view.View
+
+	ctbMetricsName = "clustertriggerbinding_count"
+	ctbCount       = stats.Float64(ctbMetricsName,
+		"number of clustertriggerbinding",
+		stats.UnitDimensionless)
+	ctbCountView *view.View
+
+	ttMetricsName = "triggertemplate_count"
+	ttCount       = stats.Float64(ttMetricsName,
+		"number of triggertemplate",
+		stats.UnitDimensionless)
+	ttCountView *view.View
+
+	ciMetricsName = "clusterinterceptor_count"
+	ciCount       = stats.Float64(ciMetricsName,
+		"number of clusterinterceptor",
+		stats.UnitDimensionless)
+	ciCountView *view.View
+)
+
+// Recorder holds information for Trigger metrics
+type Recorder struct {
+	initialized     bool
+	client          triggersclientset.Interface
```

Comment:
can we use the informers that we already have setup in the reconciler instead of having to make direct API calls?

Reply:
I changed it to Informers's listers.